pydantic/validators.py

```diff
@@ -109,6 +109,8 @@ def int_validator(v: Any) -> int:
         return v
 
     with change_exception(errors.IntegerError, TypeError, ValueError):
+        if isinstance(v, (str, bytes)):
+            return int(v, base=0)
```

Comment:
my only concern is `int('0780')` (or `-0780`) which was previously okay but would not be illegal.

I think we need to somehow allow that case.

Also, 

```py
In [1]: n = [str(n) for n in range(1000)]                                                                                                                          

In [2]: %timeit [int(v) for v in n]                                                                                                                                
95.9 µs ± 952 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [3]: %timeit [int(v, base=0) for v in n]                                                                                                                        
144 µs ± 2.64 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```
this seems to be significantly slower and `int` is a very common type with 99.99% of cases using base 10.